src/storage/store/string_chunk_data.cpp

```diff
@@ -88,6 +88,21 @@ void StringChunkData::append(ColumnChunkData* other, offset_t startPosInOtherChu
     }
 }
 
+void StringChunkData::scan(ValueVector& output, offset_t offset, length_t length) const {
+    KU_ASSERT(offset + length <= numValues);
+    if (nullData) {
+        output.setNullFromBits(nullData->getNullData()->getNullMask().getData(), offset,
+            0 /*dstOffset*/, length);
+    }
+    for (auto i = 0u; i < length; i++) {
```

Comment:
We can have a fast path when there is no nulls (`hasNoNullsGuarantee`), so we can avoid the `if` check inside a for loop.
See a reference here, though we don't need the check on `isUnfiltered()` here.
https://github.com/kuzudb/kuzu/blob/96e9f94b8a8fc398974084b2e91afdd0fca3ff27/src/include/function/unary_function_executor.h#L127